### PR TITLE
Minor improvements to `memory_resource`

### PR DESCRIPTION
**Pull request by miscco**
This removes the `cuda::get_property` customization point.

It doesnt really serve any purpose because we only ever forward to ADL. To add insult to injury it also break older compilers when we try to define `get_property` free functions for types in namespace cuda

Secondly this adds global instances of the three common memory resources. 

This allows users to utilize them without having to think about lifetime issues and with them being stateless we do not have any issues either anyway

### CODE REVIEW

**Review by bernhardmgruber — APPROVED** [review]

File: libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h [hunk 1]
```diff
@@ -155,6 +155,9 @@ struct cuda_memory_resource
 };
 static_assert(resource_with<cuda_memory_resource, device_accessible>, "");
 
+//! @brief globally available instance of a ``cuda_memory_resource``
+_LIBCUDACXX_CPO_ACCESSIBILITY cuda_memory_resource default_memory_resource{};
```
Comment (harrism) [inline libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h:159]: Suggestion: `default_memory_resource` is ambiguous.

```suggestion
_LIBCUDACXX_CPO_ACCESSIBILITY cuda_memory_resource default_cuda_memory_resource{};
```

Comment (harrism) [reply libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h:159]: Arguably should also apply to the other two.

File: libcudacxx/include/cuda/__memory_resource/cuda_managed_memory_resource.h [hunk 2]
```diff
@@ -177,6 +177,9 @@ class cuda_managed_memory_resource
 static_assert(resource_with<cuda_managed_memory_resource, device_accessible>, "");
 static_assert(resource_with<cuda_managed_memory_resource, host_accessible>, "");
 
+//! @brief globally available instance of a ``cuda_managed_memory_resource`` with flags set to ``cudaMemAttachGlobal``
```
Comment (harrism) [inline libcudacxx/include/cuda/__memory_resource/cuda_managed_memory_resource.h:180]: Question: why double back quotes?

**Review by harrism — COMMENTED** [review]
I realize that we *can* have global instances of the core memory resource types because they are stateless. But *should* we do this? It's a minor convenience. To your point about users not having to worry about lifetime, well, they can create their own global instance quite easily to solve the lifetime issue in the exact same way. 

I just worry that at some future time we will regret having these global instances for some unforeseeable reason, and we won't be able to remove them because someone is using them.

File: libcudacxx/include/cuda/__memory_resource/cuda_managed_memory_resource.h (hunk 2 above)
Comment (miscco) [reply libcudacxx/include/cuda/__memory_resource/cuda_managed_memory_resource.h:180]: That is coming from me writing too much rst docs.

In rst we need to use double quotes, but in "pure" doxygen we only need one

File: libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h (hunk 1 above)
Comment (miscco) [reply libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h]: Dropped the global instances as they indeed might give us a lot of pain later on

**Comment by miscco** [comment]
dropping in order to merge as part of #1831

**Pull request closed without merging**